float32 height
int16 id